cmd/xl-v1-healing.go

```diff
@@ -36,21 +36,41 @@ func healFormatXL(storageDisks []StorageAPI) (err error) {
 		return err
 	}
 
-	// Handles different cases properly.
-	switch reduceFormatErrs(sErrs, len(storageDisks)) {
-	case errCorruptedFormat:
-		if err = healFormatXLCorruptedDisks(storageDisks); err != nil {
+	numDisks := len(storageDisks)
+	_, unformattedDiskCount, diskNotFoundCount,
+		corruptedFormatCount, otherErrCount := formatErrsSummary(sErrs)
+
+	switch {
+	case unformattedDiskCount == numDisks:
+		// all unformatted.
+		if err = initFormatXL(storageDisks); err != nil {
+			return err
+		}
+
+	// No format heal action to take?!
+	case unformattedDiskCount == 0 && corruptedFormatCount == 0:
+		return nil
+
+	case diskNotFoundCount > 0:
```

Comment:
I am thinking that we should test this case `case diskNotFoundCount > 0:` before this `case unformattedDiskCount == 0 && corruptedFormatCount == 0:`